client/extensions/woocommerce/state/data-layer/action-list/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Internal dependencies
+ */
+import { getActionList } from 'woocommerce/state/action-list/selectors';
+import { actionListStepStart } from 'woocommerce/state/action-list/actions';
+import {
+	WOOCOMMERCE_ACTION_LIST_STEP_START,
+	WOOCOMMERCE_ACTION_LIST_STEP_SUCCESS,
+} from 'woocommerce/state/action-types';
+
+export function handleStepStart( { dispatch, getState }, action ) {
+	const { stepIndex } = action;
+
+	const actionList = getActionList( getState() );
+	const step = actionList[ stepIndex ];
```

Comment:
What use does the `stepIndex` parameter have? Don't you *always* want to execute the next step? If you calculate it using a selector, you won't need to pass it in the `action`, and you'll probably solve the `TODO: Sanity check` that's a few lines down from here.